src/components/Button.js

```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import Proptypes from "prop-types";
+
+function Button(props) {
+    const { children, icon, ...rest } = props;
+
+    return (
+        <button {...rest}>
+            {icon && <i className={icon}></i>}&nbsp;
```

Comment:
you know `fa` is common in class 
```suggestion
            {icon && <i className={`fa fa-${icon}`}></i>}&nbsp;
```
use like this